packages/cli-hydrogen/src/cli/services/deploy/deployer.test.ts

```diff
@@ -1,66 +1,123 @@
-import {healthCheck, ping} from './deployer.js'
-import {retryOnError} from './error.js'
+import {createDeployment, healthCheck, uploadDeployment} from './deployer.js'
+import {ReqDeployConfig} from './types.js'
 import {beforeEach, describe, it, expect, vi} from 'vitest'
-import {http, output} from '@shopify/cli-kit'
+import {http, api} from '@shopify/cli-kit'
+import {zip} from '@shopify/cli-kit/node/archiver'
+import {createReadStream} from 'node:fs'
+
+const defaultConfig: ReqDeployConfig = {
+  deploymentToken: '123',
+  dmsAddress: 'unit.test',
+  healthCheck: false,
+  path: '/unit/test',
+  commitMessage: 'commitMessage',
+  commitAuthor: 'commitAuthor',
+  commitSha: 'commitSha',
+  timestamp: 'timestamp',
+  commitRef: 'commitRef',
+}
 
 beforeEach(() => {
   vi.mock('@shopify/cli-kit')
-  vi.mock('./error.js', async () => {
-    const module: any = await vi.importActual('./error.js')
-    return {
-      ...module,
-      retryOnError: vi.fn(),
+  vi.mock('@shopify/cli-kit/node/archiver')
+  vi.mock('node:fs')
+})
+
+describe('createDeploymentStep()', () => {
```

Comment:
It could be nice if these were a bit more descriptive rather than just the method name (e.g. creating a deployment on DMS) or something like that.